src/main/java/com/mraof/minestuck/skaianet/UnderlingController.java

```diff
@@ -30,8 +30,12 @@ public static GristType getUnderlingType(UnderlingEntity entity)
 		ChunkGenerator<?> chunkGenerator = entity.world.isRemote ? null : ((ServerWorld)entity.world).getChunkProvider().getChunkGenerator();
 		if(chunkGenerator instanceof LandChunkGenerator)
 		{
+
 			BlockPos pos = entity.getPosition();
-			return ((LandChunkGenerator)chunkGenerator).randomLayer(entity.getRNG()).getTypeAt(pos.getX(), pos.getZ());
+			//checks whether the underling is being spawned at the entry island. If not, rely on grist layers.
+			if ((pos.getX() < entity.getEntityWorld().getDimension().getSpawnPoint().getX() + 30 && pos.getZ() < entity.getEntityWorld().getDimension().getSpawnPoint().getZ() + 30) && (pos.getX() > entity.getEntityWorld().getDimension().getSpawnPoint().getX() - 30 && pos.getZ() > entity.getEntityWorld().getDimension().getSpawnPoint().getZ() - 30))
+				return GristHelper.getPrimaryGrist(entity.getRNG());
+			else return ((LandChunkGenerator)chunkGenerator).randomLayer(entity.getRNG()).getTypeAt(pos.getX(), pos.getZ());
```

Comment:
I do not currently agree that we should go back to the old grist type generation method around spawn.
Layers has a thing in which two out of three layers has a specific grist type around 0, 0 which is also used for the cost of the punch designix. Ignoring layers and using all grist types around 0, 0 goes against that feature.